app/src/main/java/io/gnosis/safe/utils/AppUtils.kt

```diff
@@ -0,0 +1,40 @@
+package io.gnosis.safe.utils
+
+import android.app.Activity
+import android.content.Intent
+import io.gnosis.safe.qrscanner.QRCodeScanActivity
+import pm.gnosis.models.AddressBookEntry
+
+fun handleQrCodeActivityResult(
+    requestCode: Int, resultCode: Int, data: Intent?,
+    onQrCodeResult: (String) -> Unit, onCancelledResult: (() -> Unit)? = null
+): Boolean {
+    if (requestCode == QRCodeScanActivity.REQUEST_CODE) {
+        if (resultCode == Activity.RESULT_OK && data != null && data.hasExtra(QRCodeScanActivity.RESULT_EXTRA)) {
+            onQrCodeResult(data.getStringExtra(QRCodeScanActivity.RESULT_EXTRA))
+        } else if (resultCode == Activity.RESULT_CANCELED) {
+            onCancelledResult?.invoke()
+        }
+        return true
+    }
+    return false
+}
+
+fun handleAddressBookResult(
```

Comment:
Should we remove this, not sure when address book is planned, but might be a little bit in the future

Reply:
I left all the places where the logic for the address book is implemented commented out, so maybe it doesn't hurt to have this ready. I think the `AddressInputHelper` is using this method anyway, even though it will never get called in the current implementation.